lib/utilities/firebase_service.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+class FirebaseService {
+  final FirebaseFirestore _firestore = FirebaseFirestore.instance;
+
+  // Fetch all sports
+  Future<List<Map<String, dynamic>>> getSports() async {
+    try {
+      final snapshot = await _firestore.collection('sports').get();
+      return snapshot.docs.map((doc) {
+        final data = doc.data();
+        data['id'] = doc.id; // Include the document ID
+        return data;
+      }).toList();
+    } catch (e) {
+      print('Error fetching sports: $e');
+      return [];
+    }
+  }
+
+  // Fetch events for a specific sport
+  Future<List<Map<String, dynamic>>> getEventsForSport(String sportId) async {
+    try {
+      final snapshot = await _firestore
+          .collection('events-collection')
+          .where('sportId', isEqualTo: sportId)
+          .where('dateTime', isGreaterThan: Timestamp.now()) // Filter future events
+          .get();
+
+      return snapshot.docs.map((doc) {
+        final data = doc.data();
+        data['id'] = doc.id; // Include the document ID
+        return data;
+      }).toList();
+    } catch (e) {
+      print('Error fetching events for sport $sportId: $e');
+      return [];
+    }
+  }
+
+  // Register a user for an event
+  Future<void> registerForEvent(String eventId, String userId) async {
+    try {
+      final eventDoc = _firestore.collection('events-collection').doc(eventId);
+      await eventDoc.update({
+        'registeredUsers': FieldValue.arrayUnion([userId]),
+        'slotsAvailable': FieldValue.increment(-1), // Decrement available slots
+      });
+      print('User $userId registered for event $eventId');
+    } catch (e) {
+      print('Error registering for event: $e');
```

Comment:
Display an error to the user so they can try again later, you're just doing nothing.